### PR TITLE
[Cross Client Batching] Broadside LeaderTimeGetter

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**:
- Make requests to timelock efficiently

**Implementation Description (bullets)**:
- Makes requests to TimeLock for multiple leader times, based on the entire history of namespaces known to this instance of this class. The idea here is that on a stable, busy stack, the overhead of doing this is near zero (since all namespaces have requests), and we can simply re-use the existing set that was used to send the request.

**Testing (What was existing testing like?  What have you done to improve it?)**: Added a couple of tests, including simulations. Also ran the integration tests using Broadside instead of the Legacy LeaderTimeGetter and tests passed (see `arraylist` commit)

**Concerns (what feedback would you like?)**:
- Is the overhead going to be a problem?

**Where should we start reviewing?**: BLPTests

**Priority (whenever / two weeks / yesterday)**: this week

### CODE REVIEW

File: atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java [hunk 1]
```diff
@@ -1206,7 +1206,7 @@ private static LockAndTimestampServices getLockAndTimestampServices(
         LockWatchManagerImpl lockWatchManager = new LockWatchManagerImpl(lockWatchEventCache, lockWatchingService);
 
         LeaderTimeGetter leaderTimeGetter = getLeaderTimeGetter(
-                timelockNamespace, timelockRequestBatcherProviders, serviceProvider, namespacedConjureTimelockService);
+                namespacedConjureTimelockService, timelockNamespace, timelockRequestBatcherProviders, serviceProvider);
```
Comment (sudiksha27) [inline atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java:1210]: nit: unnecessary diff?

**Review by sudiksha27 — COMMENTED** [review]
LGTM! Waiting for prod validation.

**Comment by jeremyk-91** [comment]
At least on internal herby stack, this solution was not good enough.
